File: delivery/src/main/java/br/org/restaurantepopular/delivery/config/UserModule.java
package br.org.restaurantepopular.delivery.config;


import br.org.restaurantepopular.data_db.repository.user.UserDAO;
import br.org.restaurantepopular.data_db.repository.user.UserRepository;
import br.org.restaurantepopular.usecase.user.*;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;




@Configuration
public class UserModule {

    @Bean
    public GetUserByIdUseCase getUserByIdUseCase(
            IUserRepository userRepository){
        return new GetUserByIdUseCase(userRepository);

    }
    @Bean
    public CreateUserUseCase createUserUseCase(
            IUserRepository userRepository) {
        return new CreateUserUseCase(userRepository);
    }

    @Bean
    public UpdateUserUseCase updateUserUseCase(
            IUserRepository userRepository) {
        return new UpdateUserUseCase(userRepository);
    }

    @Bean
    public ListUsersUserCase listUserUseCase(
            IUserRepository userRepository) {
        return new ListUsersUserCase(userRepository);
    }

    @Bean
    public DeleteUserUseCase deleteUserUseCase(
            IUserRepository userRepository) {
        return new DeleteUserUseCase(userRepository);
    }

    @Bean
    public IUserRepository userRepository(UserDAO userDAO) {
        return new UserRepository(userDAO);
    }


}

File: delivery/src/main/java/br/org/restaurantepopular/delivery/config/UserModule.kt
package br.org.restaurantepopular.delivery.config

import br.org.restaurantepopular.data_db.repository.user.UserDAO
import br.org.restaurantepopular.data_db.repository.user.UserRepository
import br.org.restaurantepopular.usecase.user.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class UserModule {

    @Bean
    fun getUserByIdUseCase(userRepository: IUserRepository) = GetUserByIdUseCase(userRepository)

    @Bean
    fun createUserUseCase(userRepository: IUserRepository) = CreateUserUseCase(userRepository)

    @Bean
    fun updateUserUseCase(userRepository: IUserRepository) = UpdateUserUseCase(userRepository)

    @Bean
    fun listUserUseCase(userRepository: IUserRepository) = ListUsersUseCase(userRepository)

    @Bean
    fun deleteUserUseCase(userRepository: IUserRepository) = DeleteUserUseCase(userRepository)

    @Bean
    fun userRepository(userDAO: UserDAO) = UserRepository(userDAO)

}
